### PR TITLE
feat(LINGUIST-258): Sync User service profiles with ML service

### PR DESCRIPTION
**Pull request by tolgaozgun**
This makes a non-blocking call to the ML service every time user updates their profile.

### CODE REVIEW

File: src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java [hunk 1]
```diff
@@ -73,7 +96,41 @@ public RUserProfile updateUserProfile(String email, QUserProfile profile) throws
...
+                .body(Mono.just(requestBody), Map.class)
+                .retrieve()
+                .bodyToMono(String.class)
+                .subscribe(response -> {
+                    if (response != null) {
+                        logger.info("updateProfile - Response from ML service: " + response);
+                    }
+                }, error -> {
+                    if (error != null) {
+                        logger.error("updateProfile - Error from ML service: " + error.getMessage());
```
Comment (can-aslan) [inline src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java:129]: Make sure to change the logging here when you remove slf4j

Comment (tolgaozgun) [reply src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java]: Applied

File: src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java [hunk 2]
```diff
@@ -73,7 +96,41 @@ public RUserProfile updateUserProfile(String email, QUserProfile profile) throws
...
+
+            getWebClient().put()
+                .uri(ServiceUris.ML_SERVICE_UPDATE_PROFILE)
+                .header(Header.USER_EMAIL, email)
+                .body(Mono.just(requestBody), Map.class)
+                .retrieve()
+                .bodyToMono(String.class)
+                .subscribe(response -> {
+                    if (response != null) {
+                        logger.info("updateProfile - Response from ML service: " + response);
```
[the remark this answers is not in the document]
Comment (tolgaozgun) [reply src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java]: Fixed

File: src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java [hunk 3]
```diff
@@ -73,7 +96,41 @@ public RUserProfile updateUserProfile(String email, QUserProfile profile) throws
...
+            if (ML_SERVICE_BASE_URL == null || ML_SERVICE_BASE_URL.isEmpty()) {
+                logger.warn("ML Service Base URL is not set!");
+                return userProfile;
+            }
+
+            // Create request body for the request to ML service
+            Map<String, Object> requestBody = new HashMap<>();
+            requestBody.put("profile", userProfile);
+
+            getWebClient().put()
```
[the remark this answers is not in the document]
Comment (tolgaozgun) [reply src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java]: Fixed

File: src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
 import java.util.List;
+import java.util.Map;
 import java.util.UUID;
 
+import app.linguistai.bmvp.consts.Header;
+import app.linguistai.bmvp.consts.ServiceUris;
 import app.linguistai.bmvp.repository.IUserHobbyRepository;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
[the remark this answers is not in the document]
Comment (tolgaozgun) [reply src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java]: Fixed

File: src/main/java/app/linguistai/bmvp/consts/ServiceUris.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class ServiceUris {
+
+
+    public static final String ML_SERVICE_PROFILE_APP = "/profile";
+
+    public static final String ML_SERVICE_UPDATE_PROFILE = ML_SERVICE_PROFILE_APP + "/update";
+}
```
[the remark this answers is not in the document]
Comment (tolgaozgun) [reply src/main/java/app/linguistai/bmvp/consts/ServiceUris.java:9]: Fixed

**Review by can-aslan — APPROVED** [review]
LGTM!

**Review by elif-krvan — APPROVED** [review]
okeyyyyyyy

**Pull request merged**